core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -527,7 +527,9 @@ private[spark] class BlockManager(
       logInfo(s"external shuffle service port = $externalShuffleServicePort")
       shuffleServerId = BlockManagerId(executorId, blockTransferService.hostName,
         externalShuffleServicePort)
-      if (!isDriver) {
+      // Do not try to register with the external shuffle server under testing since none of
+      // test modes (local, local-cluster) enables shuffle service.
+      if (!isDriver && !Utils.isTesting) {
```

Comment:
This looks orthogonal. Shall we make a new JIRA and merge independently?

Reply:
Without this, the test can't run due to the `BlockManager` creation will fail on this.

Reply:
It turns out some tests rely on this creation failure for testing purposes. I added a separate testing conf (`spark.testing.skipESSRegister`) to keep the default behavior.